src/main/java/configurationslicing/promotedbuilds/PromotedBuildsNameSlicer.java

```diff
@@ -0,0 +1,156 @@
+
+            for (AbstractProject project: allProjects) {
+                if (project instanceof Project || project instanceof MatrixProject) {
+                    filteredProjects.add(project);
+                }
+            }
+            return filteredProjects;
+        }
+
+        @Override
+        public List<String> getValues(AbstractProject<?,?> item) {
+
+            JobPropertyImpl property = item.getProperty(JobPropertyImpl.class);
+
+            List<String> valuesList = new ArrayList<String>();
+
+            if (property != null) {
+                String nameString = "";
+                for(PromotionProcess process : property.getActiveItems()) {
+                    nameString = (String.format("%s[%s] ", nameString, process.getName()));
+                }
+                valuesList.add(nameString.trim());
+            }
+            if (valuesList.isEmpty() || valuesList.get(0).equals("")) {
+                valuesList.add(NOTHING);
+            }
+
+            return valuesList;
+        }
+
+        @Override
+        public boolean setValues(AbstractProject<?,?> item, List<String> rawValues) {
+
+            JobPropertyImpl property = item.getProperty(JobPropertyImpl.class);
+
+            List<String> values = parseValues(rawValues);
+
+            if (property == null && !values.isEmpty()) {
+                property = tryCreateProperty(item);
+                if (property == null) {
+                    return false;
+                }
+            }
+
+            if(rawValues.get(0).equals(NOTHING)) {
+                property.getActiveItems().removeAll(property.getActiveItems());
+            }
+
+            Map<String, PromotionProcess> oldPromotions = new HashMap<String, PromotionProcess>();
+            for(PromotionProcess promotionProcess : property.getActiveItems()) {
+                oldPromotions.put(promotionProcess.getName(), promotionProcess);
+            }
+
+            // Remove old promotions
+            for(String oldName : oldPromotions.keySet()) {
+                if(!values.contains(oldName)) {
+                    property.getActiveItems().remove(oldPromotions.get(oldName));
+                }
+            }
+
+            // Add new promotions
+            for(String newName : values) {
+                if(!oldPromotions.keySet().contains(newName)) {
+                    try {
+                        property.addProcess(newName);
+                    } catch (IOException e) {
+                        log.error("Failed to add promotion process {} to project {}", newName, item.getName());
+                        Throwables.propagate(e);
+                        return false;
+                    }
+                }
+            }
+
+            return true;
+        }
+
+        private List<String> parseValues(List<String> rawValues) {
+
+            List<String> values = new ArrayList<String>();
+
+            Pattern regex = Pattern.compile("\\[(.*?)\\]");
```

Comment:
Name this pattern by what it does.

Also why do we still need this pattern? I thought we don't need brackets for deletions any more.